File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> counts;
        for (int num : nums) {
            counts[num]++;
        }
        vector<pair<int, int>> pairs {};
        for (auto& p : counts) {
            pairs.push_back({p.second, p.first});
        }
        sort(pairs.rbegin(), pairs.rend());
        vector<int> result;
        for (int i = 0; i < k; i++)
            result.push_back(pairs[i].second);
        return result;
    }
};

File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        pairs = {
            ")" : "(", 
            "]" : "[", 
            "}" : "{", 
        }

        stack = []

        for c in s:
            if c in ["(", "[", "{"]:
                stack.append(c)
            if c in pairs.keys():
                if len(stack) or pairs[c] != stack[-1]:
                    return False
                else:
                    stack.pop()
        
        if len(stack):
            return False

        return True